src/ngraph/op/deformable_convolution.cpp

```diff
@@ -0,0 +1,177 @@
+
+op::v1::DeformableConvolution::DeformableConvolution(const Output<Node>& arg,
+                                                     const Output<Node>& deformable_values,
+                                                     const Output<Node>& filters,
+                                                     const Strides& strides,
+                                                     const CoordinateDiff& pads_begin,
+                                                     const CoordinateDiff& pads_end,
+                                                     const Strides& dilations,
+                                                     const PadType& auto_pad,
+                                                     const size_t group,
+                                                     const size_t deformable_group)
+    : Op({arg, deformable_values, filters})
+    , m_strides(strides)
+    , m_dilations(dilations)
+    , m_pads_begin(pads_begin)
+    , m_pads_end(pads_end)
+    , m_auto_pad(auto_pad)
+    , m_group(group)
+    , m_deformable_group(deformable_group)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::DeformableConvolution::validate_and_infer_types()
+{
+    const PartialShape& data_batch_shape = get_input_partial_shape(0);
+    const PartialShape& deformable_values_shape = get_input_partial_shape(1);
+    const PartialShape& filters_shape = get_input_partial_shape(2);
+
+    element::Type data_batch_et = get_input_element_type(0);
+    element::Type deformable_values_et = get_input_element_type(1);
+    element::Type filters_et = get_input_element_type(2);
+
+    if (m_strides.size() == 0)
+    {
+        m_strides = conv_default_strides(this, data_batch_shape, filters_shape);
+    }
+
+    if (m_dilations.size() == 0)
+    {
+        m_dilations = conv_default_strides(this, data_batch_shape, filters_shape);
+    }
+
+    if (m_pads_begin.size() == 0)
+    {
+        m_pads_begin = conv_default_padding(this, data_batch_shape, filters_shape);
+    }
+
+    if (m_pads_end.size() == 0)
+    {
+        m_pads_end = conv_default_padding(this, data_batch_shape, filters_shape);
+    }
+
+    if (m_auto_pad == PadType::SAME_UPPER || m_auto_pad == PadType::SAME_LOWER)
+    {
+        if (data_batch_shape.is_static() && filters_shape.is_static())
+        {
+            m_pads_begin.clear();
+            m_pads_end.clear();
+            auto filter_shape = filters_shape.to_shape();
+            filter_shape.erase(filter_shape.begin(), filter_shape.begin() + 2); // Remove {O,I}
+            infer_auto_padding(data_batch_shape.to_shape(),
+                               filter_shape,
+                               m_strides,
+                               m_dilations,
+                               m_auto_pad,
+                               m_pads_end,
+                               m_pads_begin);
+        }
+    }
+
+    if (deformable_values_shape.is_static())
```

Comment:
I think that better (less restrictive) is using `deformable_values_shape.rank()` instead of `deformable_values_shape.to_shape().size()`